src/snowflake/snowpark/modin/plugin/extensions/index.py

```diff
@@ -501,12 +501,20 @@ def astype(self, dtype: Any, copy: bool = True) -> Index:
         >>> idx.astype('float')
         Index([1.0, 2.0, 3.0], dtype='float64')
         """
-        WarningMessage.index_to_pandas_warning("astype")
-        return Index(
-            self.to_pandas().astype(dtype=dtype, copy=copy),
-            dtype=dtype,
-            convert_to_lazy=self.is_lazy,
-        )
+        if dtype is not None:
+            dtype = pandas_dtype(dtype)
+
+        if self.dtype == dtype:
+            # Ensure that self.astype(self.dtype) is self
+            return self.copy() if copy else self
+
+        col_dtypes = {
+            column: pandas_dtype(dtype)
```

Comment:
nit: We already converted to pandas_dtype above. This can just be `column: dtype`